src/var.c

```diff
@@ -53,8 +55,37 @@ struct fi_var_t {
 
 static struct fi_var_t *fi_vars = NULL;
 
+__attribute__((visibility ("default")))
+int DEFAULT_SYMVER_PRE(fi_getvars)(struct fi_env **vars, int *count) {
+
+	struct fi_env *vhead;
+	int len = 0;
+
+	// just get a count
+	for (struct fi_var_t *ptr = fi_vars; ptr ;ptr = ptr->next, ++len) {}
+
+	vhead = calloc(len, sizeof (*vhead));
+
+	if (!vhead)
+		return -FI_ENOMEM;
+
+	int i = 0;
+	for (struct fi_var_t *ptr = fi_vars; ptr; ptr = ptr->next, ++i) {
+		vhead[i] = (struct fi_env) {	.provider = ptr->provider,
+						.var_name = ptr->var_name,
+						.env_var_name = ptr->env_var_name,
+						.help_string = ptr->help_string
+					   };
```

Comment:
I could do either style for assignment, I am not tied to either one.

Racy API: yes two threads can have an inconsistent view of the environment vars, but adding locking won't change this right?

This returns pointers to internal structures but the only memory alloc'd is the storage to store the pointers, so:

```
fi_free_vars(struct fi_env *env) { 
    free(env);
}
```

Won't free any internal vars and is all we need right?